### PR TITLE
feat(scheduler): implement NodePreferAvoidPods as score plugin

### LINKED ISSUE #83770: [migration phase 1] NodePreferAvoidPods as Score plugin
**What would you like to be added**:

Add a Score plugin that calls into NodePreferAvoidPods priority, example #83601. Score Plugin name should be the same.

**Why is this needed**:

Part of  #83554

/sig scheduling
/help
/priority important-soon

### PR DESCRIPTION
**Pull request by draveness**
/kind cleanup
/assign @ahg-g

**Which issue(s) this PR fixes**:

Fixes #83770

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by ahg-g — COMMENTED** [review]
/lgtm
/approve

File: pkg/scheduler/framework/plugins/nodepreferavoidpods/node_prefer_avoid_pods_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+			p, _ := New(nil, fh)
+			var gotList framework.NodeScoreList
+			for _, n := range test.nodes {
+				nodeName := n.ObjectMeta.Name
+				score, status := p.(framework.ScorePlugin).Score(state, test.pod, nodeName)
+				if !status.IsSuccess() {
+					t.Errorf("unexpected error: %v", status)
+				}
+				gotList = append(gotList, framework.NodeScore{Name: nodeName, Score: score})
+			}
```
[the remark this answers is not in the document]
Comment (draveness) [reply pkg/scheduler/framework/plugins/nodepreferavoidpods/node_prefer_avoid_pods_test.go]: sure, will do

**Comment by ahg-g** [comment]
/hold cancel

since the other one is on hold now.

**Pull request merged**